term/fhir-term/src/main/java/com/ibm/fhir/term/service/FHIRTermService.java

```diff
@@ -1167,29 +1166,77 @@ private List<FHIRTermServiceProvider> loadProviders() {
         return providers;
     }
 
-    private ValidationOutcome validateDisplay(CodeSystem codeSystem, Coding coding, LookupOutcome lookupOutcome) {
+    /**
+     * Validate a coding display against the passed lookupOutcome, inferring case sensitivity from the registry
+     *
+     * @param coding
+     *     the coding
+     * @param lookupOutcome
+     *     the outcome of a CodeSystem lookup
+     * @return
+     *     the outcome of validation; true if either the lookupOutcome or coding parameters are null or have a null display
+     * @implNote this method looks up the implied CodeSystem from the FHIR registry to determine case sensitivity,
+     *     then delegates to {@link #validateDisplay(Coding, LookupOutcome, boolean)}
+     */
+    private ValidationOutcome validateDisplay(Coding coding, LookupOutcome lookupOutcome) {
+        if (coding == null) {
+            return buildValidationOutcome(true, null, lookupOutcome);
+        }
+
+        java.lang.String system = (coding.getSystem() != null) ? coding.getSystem().getValue() : null;
+        java.lang.String version = (coding.getVersion() != null) ? coding.getVersion().getValue() : null;
+
+        boolean caseSensitive = false;
+        if (system != null) {
+            java.lang.String url = (version != null) ? system + "|" + version : system;
+            caseSensitive = CodeSystemSupport.isCaseSensitive(url);
+        }
+
+        return validateDisplay(coding, lookupOutcome, caseSensitive);
+    }
+
+    /**
+     * Validate a coding display against the passed lookupOutcome
+     *
+     * @param coding
+     *     the coding
+     * @param lookupOutcome
+     *     the outcome of a CodeSystem lookup
+     * @param isCaseSensitive
+     *     whether the underlying CodeSystem is case-sensitive or not
+     * @return
+     *     the outcome of validation; true if either the lookupOutcome or coding parameters are null or have a null display
+     */
+    private ValidationOutcome validateDisplay(Coding coding, LookupOutcome lookupOutcome, boolean isCaseSensitive) {
         if (lookupOutcome == null || coding == null ||
                 lookupOutcome.getDisplay() == null || coding.getDisplay() == null ||
                 lookupOutcome.getDisplay().getValue() == null && coding.getDisplay().getValue() == null) {
             return buildValidationOutcome(true, null, lookupOutcome);
         }
 
-        java.lang.String system = null;
-        if (coding.getSystem() != null) {
-            system = coding.getSystem().getValue();
-        } else if (codeSystem != null && codeSystem.getUrl() != null) {
-            system = codeSystem.getUrl().getValue();
+        java.lang.String displayToValidate = isCaseSensitive ? coding.getDisplay().getValue() : normalize(coding.getDisplay().getValue());
+
+        boolean result = false;
```

Comment:
Minor. No need to assign a default value. But I like to avoid assigning a default because it helps to make sure a value gets assigned in all possible branches of the subsequent code.

Reply:
so i usually assign false to be explicit / help me remember it'll have a value

Reply:
If they are member variables they are assigned false by default, but variables in a method do not get assigned a default value.